middle_end/flambda/terms/cost_metrics.rec.ml

```diff
@@ -66,17 +66,25 @@ and named_size ~find_code (named : Named.t) size =
   | Set_of_closures set_of_closures ->
     let func_decls = Set_of_closures.function_decls set_of_closures in
     let funs = Function_declarations.funs func_decls in
+    let num_clos_vars =
+      Set_of_closures.closure_elements set_of_closures
+      |> Var_within_closure.Map.cardinal
+    in
     Closure_id.Map.fold (fun _ func_decl size ->
       let code_id = Function_declaration.code_id func_decl in
       let code = find_code code_id in
-      match Code.params_and_body code with
-      | Present params_and_body ->
-        Function_params_and_body.pattern_match params_and_body
-          ~f:(fun ~return_continuation:_ _exn_continuation _params
-               ~body ~my_closure:_ ~is_my_closure_used:_ ->
-               expr_size ~find_code body size)
-      | Deleted -> size)
-      funs size
+      let arity = List.length (Code.params_arity code) in
+      let size =
+        match Code.params_and_body code with
+        | Present params_and_body ->
+          Function_params_and_body.pattern_match params_and_body
+            ~f:(fun ~return_continuation:_ _exn_continuation _params
+                 ~body ~my_closure:_ ~is_my_closure_used:_ ->
+                 expr_size ~find_code body size)
+        | Deleted -> size
+      in
+      size + (if arity = 1 then 2 else 3))
+      funs (Code_size.to_int Code_size.alloc_size + num_clos_vars + size)
```

Comment:
Please indent the code above by 2 spaces, so it is distinguished from the remaining arguments to `fold`.